graviti/operation/frame.py

```diff
@@ -89,6 +94,39 @@ def _get_max_batch_size(self) -> int:
             _MAX_ITEMS // self._data.schema._get_column_count(),  # pylint: disable=protected-access
         )
 
+    def _update_local_files(
+        self,
+        object_policy_manager: "ObjectPolicyManager",
+    ) -> List[List[Optional[Tuple[str, Path]]]]:
+        # object_policy_manager = self._data.object_policy_manager
+        if not object_policy_manager:
+            raise ValueError("Require object policy manager to upload data")
+
+        # pylint: disable=protected-access
+        prefix = object_policy_manager._init_put_policy()["policy"][  # type: ignore[attr-defined]
+            "prefix"
+        ]
+
+        keys_and_paths = []
+        for file_array in self.get_file_arrays():
+            schema = file_array.schema
+            file_type = RemoteFileTypeResgister.SCHEMA_TO_REMOTE_FILE[
+                schema.package.repo, schema.__class__.__name__  # type: ignore[index]
+            ]
+            key_and_path: List[Optional[Tuple[str, Path]]] = []
+            for index, file in enumerate(file_array._data):
+                if isinstance(file, File):
+                    key = f"{prefix}/{file.get_checksum()}"
+                    key_and_path.append((key, file.path))
+
+                    post_data = file._to_post_data()
+                    post_data["key"] = key
+                    file_array[index] = file_type(**post_data)  # type: ignore[arg-type]
```

Comment:
the file_array comes from a copied DataFrame, update file_array will not effect the sheet.